src/component/panels/extra/utilities/ZoneUtilities.js

```diff
@@ -51,4 +59,89 @@ const unlink = (zone, isOnZoneLevel, signalIndex, axis) => {
   return zoneObject;
 };
 
-export { getDiaIDs, getPubIntegral, resetDiaIDs, unlink };
+const unlinkInAssignmentData = (
+  assignmentData,
+  zone,
+  isOnZoneLevel,
+  signalIndex,
+  axis,
+) => {
+  if (isOnZoneLevel !== undefined && axis !== undefined) {
+    if (isOnZoneLevel === true) {
+      assignmentData.dispatch({
+        type: 'REMOVE_ALL',
+        payload: {
+          id: zone.id,
+          axis: axis,
+        },
+      });
+    } else if (signalIndex !== undefined) {
+      assignmentData.dispatch({
+        type: 'REMOVE_ALL',
+        payload: {
+          id: `${zone.id}${HighlightSignalConcatenation}${signalIndex}`,
+          axis: axis,
+        },
+      });
+    }
+  } else if (axis !== undefined) {
+    assignmentData.dispatch({
+      type: 'REMOVE_ALL',
+      payload: {
+        id: zone.id,
+        axis: axis,
+      },
+    });
+    zone.signal.forEach((_signal, i) => {
+      assignmentData.dispatch({
+        type: 'REMOVE_ALL',
+        payload: {
+          id: `${zone.id}${HighlightSignalConcatenation}${i}`,
+          axis: axis,
+        },
+      });
+    });
+  } else {
+    assignmentData.dispatch({
+      type: 'REMOVE_ALL',
+      payload: {
+        id: zone.id,
+        axis: 'x',
+      },
+    });
+    zone.signal.forEach((_signal, i) => {
+      assignmentData.dispatch({
+        type: 'REMOVE_ALL',
+        payload: {
+          id: `${zone.id}${HighlightSignalConcatenation}${i}`,
+          axis: 'x',
+        },
+      });
+    });
+    assignmentData.dispatch({
+      type: 'REMOVE_ALL',
+      payload: {
+        id: zone.id,
+        axis: 'y',
+      },
+    });
+    zone.signal.forEach((_signal, i) => {
+      assignmentData.dispatch({
+        type: 'REMOVE_ALL',
+        payload: {
+          id: `${zone.id}${HighlightSignalConcatenation}${i}`,
+          axis: 'y',
+        },
+      });
+    });
+  }
+};
```

Comment:
this logic must be changed and think in a way that you have have one dispatch function and only pass different object but not use dispatch function several time, it's redundancy and not useful at all.